app/services/magic_link_flow.rb

```diff
@@ -0,0 +1,26 @@
+class MagicLinkFlow
+  def self.sign_up(candidate)
+    magic_link_token = MagicLinkToken.new
+    candidate.update(magic_link_token: magic_link_token.encrypted)
+    link = "http://localhost:3000/personal-details/new?token=#{magic_link_token.raw}"
```

Comment:
Discussed IRL: the mailer can put the URL in for us if we use a `_url` helper in the view